### DISCUSSION TITLE
How to monitor and count documentation errors?

### DISCUSSION THREAD

**Discussion by 1ucian0** [Qiskit/qiskit.org#2670] [General]
Hi all!
      On of the problems in qiskit.org/documentation is the amount of deprecation warnings and errors  like [this one](https://github.com/Qiskit/qiskit-terra/pull/8106). I'm trying to find a way to keep an eye on them.

The options so far:
     
### piggybacking google

@y4izus had the idea of using google for finding errors. For example searching for `site:qiskit.org/documentation "deprecationwarning:"`

On the bad side, this output is a bit noisy as it also returns false positives, such as [when the string `DeprecationWarning` is in source code](https://qiskit.org/documentation/_modules/qiskit/providers/aer/extensions/snapshot.html), because the `:` is being ignored. Also, it needs some time for cache updating after deploying. 

On the good side, it is fast and simple. The google bot manages to capture DOM modified by javascript and  [this](https://github.com/Qiskit/qiskit-terra/pull/8106) is [detectable](https://www.google.com/search?q=site%3Aqiskit.org%2Fdocumentation+%22misplaced+%26%22). 

### web scraping

This script, mirrors the website and allows for searches in the html dump:
```sh
#!/bin/bash

HOST=https://qiskit.org/

DIRS="documentation documentation/dynamics/"

lftp ${HOST} << EOF
mirror --use-pget-n=10 ${DIRS} > errors.list
bye
EOF
echo "done"

grep -r 'DeprecationWarning:' documentation/ > DeprecationWarning.list

echo "errors"
cat errors.list | wc -l
echo "Deprecations"
cat DeprecationWarning.list | wc -l
```
However, does not detect the `Misplaced &` issue. 

Any other idea on how to monitor documentation errors?

**Comment by pandasa123** [comment]
I wonder if this could be used for the Textbook itself as well, which are mostly Jupyter notebooks? It would solve a common, frustrating problem that professors face
